Comm  drivers: refactor, handle missing input events and nmea 0183 v4 tags

leamas opened the pull request:
Overhaul of the 0183 handling

  - Centralize the basic n0183 checks for being parsable etc. to the common N0163 driver base.
  - Messages which are not forwarded (filtered, junk, bad checksums) are instead sent to a special EventVar in the driver registry. Data Monitor listens to this to be able to present also these input events
  - Messages with leading tags are silently stripped so only the basic sentence is forwarded.
  - Skip needless input filter tests in the multiplexer.

Closes: #4692 
Closes: #4694

leamas commented:
Rebased: bugfixes, rebased to master

Hakansv commented:
This is good I think. Solves the VDR issue.
Since I not fully understand the code two Q:
1. The tag part is abandoned as discussed, great. Should we consider extracting it and offering it to the plugin API. You never know what a plugin developer might come up with?  If so a future request?
2. Earlier we introduced the logic to accept a message without checksum. But if it's there it must be correct. Will that logic still apply with this code?

leamas commented:
>  Should we consider extracting it and offering it to the plugin API. You never know what a plugin developer might come up with? If so a future request?

Perhaps. But for now, we need the dust to settle first, there is more refactoring under way after this one.

> arlier we introduced the logic to accept a message without checksum. But if it's there it must be correct. Will that logic still apply with this code?

... but that was never documented, I presume (?). It's hard to code without specs. As it is now it accepts  messages without checksum. Messages with bad checksums re marked as such in the Data Monitor but otherwise accepted. Changing this is a simple oneliner, so it's basically aboit documenting  how it should be done.

leamas commented:
... and document this. Will do

leamas commented:
Rebased to current master
  - Made messages with bad checksum being discarded (but still visible in the DM)
  - Bugfixes
  - Added docs
  - Tested with @transmitterdan 's eminent VDRPlayer script.

leamas commented:
Rebased to current master:
  - Fix comm_drv_factory file level comments
  - Handle duplicate symbols linkage errors on useless dashboard nmea0183 library

Hakansv commented:
Testing the "Handle duplicate symbols linkage errors on useless dashboard nmea0183 library" gives the same Win error as the ci build.
btw. Q: how does this commit connect to "_handle missing input events and nmea 0183 v4 tags_"

leamas commented:
> . Q: how does this commit connect to "handle missing input events and nmea 0183 v4 tags"

Basically not at all, but required for Linux build.

Merged as 783013308. Closing
